src/murfey/server/api/bootstrap.py

```diff
@@ -565,6 +570,388 @@ def get_msys2_package_file(
+    the 'api' key points to an API version of that same registry. Both will be
+    used by Cargo when searching for and downloading packages.
+    """
+
+    # Construct URL for Rust router
+    base_url = f"{request.url.scheme}://{request.url.netloc}" + rust.prefix
+
+    # Construct config file with the necessary endpoints
+    config = {
+        "dl": f"{base_url}/crates",
+        "api": f"{base_url}",
+    }
+
+    # Save it as a JSON and return it as part of the response
+    json_data = json.dumps(config, indent=4)
+    json_bytes = io.BytesIO(json_data.encode("utf-8"))
+
+    return StreamingResponse(
+        json_bytes,
+        media_type="application/json",
+        headers={"Content-Disposition": "attachment; filename=config.json"},
+    )
+
+
+@rust.get("/index/{c1}/{c2}/{package}", response_class=StreamingResponse)
+def get_index_package_metadata(
+    request: Request,
+    c1: str,
+    c2: str,
+    package: str,
+):
+    """
+    Download the metadata for a given package from the crates.io sparse index.
+    The path to the metadata file on the server side takes the following form:
+    /{c1}/{c2}/{package}
+
+    c1 and c2 are 2 characters-long strings that are taken from the first 4
+    characters of the package name (a-z, A-Z, 0-9, -, _). For 3-letter packages,
+    c1 = 3, and c2 is the first character of the package.
+    """
+
+    logger.debug(f"Received request to access {str(request.url)}")
+
+    # Validate path to the package metadata
+    if not all(bool(re.fullmatch(r"[\w\-]{1,2}", char)) for char in (c1, c2)):
+        raise ValueError("Invalid path to package metadata")
+
+    if len(c1) == 1 and not c1 == "3":
+        raise ValueError("Invalid path to package metadata")
+    if c1 == "3" and not len(c2) == 1:
+        raise ValueError("Invalid path to package metadata")
+
+    if not bool(re.fullmatch(r"[\w\-]+", package)):
+        raise ValueError("Invalid package name")
+
+    # Request and return the metadata as a JSON file
+    url = f"{rust_index}/{c1}/{c2}/{package}"
+    response = http_session.get(url)
+    if response.status_code != 200:
+        raise HTTPException(status_code=response.status_code)
+    return StreamingResponse(
+        response.iter_content(chunk_size=8192),
+        media_type="application/json",
+    )
+
+
+@rust.get("/index/{n}/{package}", response_class=StreamingResponse)
+def get_index_package_metadata_for_short_package_names(
+    request: Request,
+    n: str,
+    package: str,
+):
+    """
+    The Rust sparse index' naming scheme for packages with 1-2 characters is
+    different from the standard path convention. They are stored under
+    /1/{package} or /2/{package}.
+    """
+
+    logger.debug(f"Received request to access {str(request.url)}")
+
+    # Validate path to crate
+    if n not in ("1", "2"):
+        raise ValueError("Invalid path to package metadata")
+    if not bool(re.fullmatch(r"[\w\-]{1,2}", package)):
```

Comment:
I wouldn't expect this `bool` cast to be necessary. You could do a `if  re.fullmatch(r"[\w\-]{1,2}", package) is None` to be safe